ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/431__data-science-pipelines-api.robot

```diff
@@ -40,6 +40,37 @@ Verify Ods Users Can Do Http Request That Must Be Redirected to Https
     Should Start With    ${url}    https
     Remove Pipeline Project    project-redirect-http
 
+Verify DSPO Operator Reconciliation Retry
+    [Documentation]    Verify DSPO Operator is able to recover from missing components during the initialization
+    [Tags]      Sanity    Tier1
+    ${local_project_name} =    Set Variable    recon-test
+    ${stopped} =    Set Variable    ${False}
+    # limit is 180 because the reconciliation run every 2 minutes
+    ${timeout} =    Set Variable    180
+    New Project    ${local_project_name}
+    Install DataSciencePipelinesApplication CR    ${local_project_name}    data-science-pipelines-reconciliation.yaml    False
+    # wait the CR be ready
+    Wait Until Keyword Succeeds    15 times    1s
+    ...    Run    oc get datasciencepipelinesapplications -n ${local_project_name}
+    ${pod_name} =    Run    oc get pods -n ${APPLICATIONS_NAMESPACE} | grep data-science-pipelines-operator | awk '{print $1}'
+    Log    ${pod_name}
+    TRY
+        WHILE    not ${stopped}    limit=${timeout}
+            Sleep    1s
+            ${logs}=    Oc Get Pod Logs
+            ...    name=${pod_name}
+            ...    namespace=${APPLICATIONS_NAMESPACE}
+            ${stopped} =    Set Variable If    "Encountered error when parsing CR" in """${logs}"""    True    False
+        END
+    EXCEPT    WHILE loop was aborted    type=start
+        Log    Reconciliation wasn't triggered
+    END
```

Comment:
you sure when the loops finishes it goes in the EXCEPT? I don't think it raise an exception or error when it exits.

Reply:
With the latest fix if something went wrong the test will fail
![image](https://github.com/red-hat-data-services/ods-ci/assets/687311/9988a155-5e90-45c6-b4d3-e4fb34158238)